src/sentry/sentry_metrics/querying/units.py

```diff
@@ -60,15 +61,19 @@ class Unit:
     def convert(self, value: float | int) -> float | int:
         return value * self.scaling_factor
 
-    def apply_on_timeseries(self, timeseries: Timeseries) -> Timeseries | Formula:
+    def apply_on_query_expression(self, query_expression: QueryExpression) -> QueryExpression:
```

Comment:
I have generalized this, since it now supports also applying the scaling factor on scalars.